epicure-admin/src/app/components/generic-table/generic-table.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-generic-table',
+  templateUrl: './generic-table.component.html',
+  styleUrls: ['./generic-table.component.scss']
+})
+export class GenericTableComponent implements OnInit {
+  @Input() pageTitle: string = '';
+  @Input() data: any[] = [];
+  @Input() columns: string[] = [];
+  @Input() columnDefs: { [key: string]: string } = {};
+
+  displayedColumns: string[] = [];
+  dataSource = new MatTableDataSource(this.data);
+  
+  constructor(private router: Router) {}
+  ngOnInit() {
+    this.displayedColumns = this.columns;
+    console.log('pageTitle', this.pageTitle);
+    console.log('Columns:', this.columns);
+    console.log('ColumnDefs:', this.columnDefs);
+    console.log('Data:', this.data);
+  }
+  
+  getElementVal(element: any, column: any) {
+      if (column === 'dishes') {
+        let dishesStr: string = '';
+        for (let index = 0; index < element.dishes.length; index++) {
+          dishesStr = dishesStr + element.dishes[index].title + ' ';
+        }
+        return dishesStr == ''?  'No dishes info': dishesStr;
+
+      }else if (column === 'chef') {
+        return element.chef?.title || 'No chefs info';  
+      
+      }else if (column === 'restaurants') {
+        let restaurantsStr: string = '';
+        for (let index = 0; index < element.restaurants.length; index++) {
+          restaurantsStr = restaurantsStr + element.restaurants[index].title + ' ';
+        }
+        return restaurantsStr == ''?  'No dishes info': restaurantsStr;
```

Comment:
you can ask restaurantsStr ? (empty string counts as false)

Reply:
not working, if I'm doing that  No restaurants info selected every time